src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/AspNetCore/ILoggerIntegration.cs

```diff
@@ -0,0 +1,86 @@
+// <copyright file="ILoggerIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.AspNetCore;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.DuckTyping;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNetCore
+{
+    /// <summary>
+    /// Microsoft.AspNetCore.Hosting.Internal.HostingApplication.ProcessRequestAsync calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "Microsoft.AspNetCore.Hosting",
+        TypeName = "Microsoft.AspNetCore.Hosting.Internal.HostingApplication",
+        MethodName = "ProcessRequestAsync",
+        ReturnTypeName = ClrNames.Task,
+        ParameterTypeNames = new[] { "Context" },
+        MinimumVersion = "2.0.0",
+        MaximumVersion = "2.*.*",
+        IntegrationName = IntegrationName)]
+    [InstrumentMethod(
+        AssemblyName = "Microsoft.AspNetCore.Hosting",
+        TypeName = "Microsoft.AspNetCore.Hosting.HostingApplication", // different namespace
+        MethodName = "ProcessRequestAsync",
+        ReturnTypeName = ClrNames.Task,
+        ParameterTypeNames = new[] { "Context" },
+        MinimumVersion = "3.0.0",
+        MaximumVersion = "5.*.*",
+        IntegrationName = IntegrationName)]
+    public class ILoggerIntegration
+    {
+        private const string IntegrationName = nameof(IntegrationIds.AspNetCore);
+        private static readonly Type ILoggerType;
+
+        static ILoggerIntegration()
+        {
+            ILoggerType = Type.GetType("Microsoft.Extensions.Logging.ILogger, Microsoft.Extensions.Logging.Abstractions", throwOnError: false);
+        }
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TContext">Type of the controller context</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="context">The context of the request</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget, TContext>(TTarget instance, TContext context)
+            where TTarget : IHostingApplication
+        {
+            IDisposable disposable = null;
+            if (Tracer.Instance.Settings.LogsInjectionEnabled)
+            {
+                var logger = (ILogger)DuckType.Create(typeof(ILogger), instance.Diagnostics.Logger, ILoggerType);
```

Comment:
Theoretically, this shouldn't be necessary, but if we make instance.Diagnostics an `ILogger` (instead of an `object`) and use chaining, then it fails. I can't work out why...

Reply:
Figured it out - it's due to the implementation in ASP.NET Core, which uses explicit interface implementation: 

https://github.com/dotnet/extensions/blob/v2.1.28/src/Logging/Logging.Abstractions/src/LoggerOfT.cs#L32-L35

Our DuckTyping fails on that currently.

Reply:
Thanks to #1555, we can now use the simpler dock-type chaining approach (done in https://github.com/DataDog/dd-trace-dotnet/pull/1549/commits/0557c1aa8f5625963cb30905e10f8a2e5154fda6)

Reply:
I think you’re missing the wildcard: https://github.com/DataDog/dd-trace-dotnet/blob/master/test/Datadog.Trace.DuckTyping.Tests/DuckExplicitInterfaceTests.cs#L50

Reply:
Can we resolve this thread?

Reply:
Yep, the problem should be resolved by https://github.com/DataDog/dd-trace-dotnet/pull/1561, so going to rebase and check